lib/fluent/plugin/out_datadog.rb

```diff
@@ -44,58 +43,36 @@ def configure(conf)
     super
   end
 
-  def client
-    @_socket ||= if @use_ssl
-      context    = OpenSSL::SSL::SSLContext.new
-      socket     = TCPSocket.new @host, @ssl_port
-      ssl_client = OpenSSL::SSL::SSLSocket.new socket, context
-      ssl_client.connect
+  def new_client
+    if @use_ssl
+      begin
+        context    = OpenSSL::SSL::SSLContext.new
+        socket     = TCPSocket.new @host, @ssl_port
+        ssl_client = OpenSSL::SSL::SSLSocket.new socket, context
+        ssl_client.connect
+        return ssl_client
+      rescue => e
+        log.warn "Could not open a secure connection to Datadog, error=#{e}"
+        sleep 2
```

Comment:
could it be a problem to automatically retry here?

Reply:
Not really but it's not really relevant since the retry logic is already handled in `send_to_datadog`, I'll remove this piece of code.